_posts/2019-01-02-Deviceid-Reliability.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: post
+title: Keep the quality of your data high
+subtitle: All that glitters is not gold
+category: data
+tags: [adtech, howto]
+author: Alessio Borgheresi
+header-img: "images/Deviceid-Reliability/data_quality.jpg"
+---
+
+At Adello we are processing a lot of data. 
+In a perfect world, we could just take them and use them directly for the benefit of our customers. 
+In reality, the data are not always correct and to produce some reliable outputs it is necessary to filter out those inconsistency.
+In this post we describe an example of the quality selection we perform at Adello, underlining the benefits we gain from it.
+
+In a [previous post](https://adello.github.io/Tracking/) we introduced the way we identify a device for advertising targeting purpose.
+In summary, we use the Android **[Advertising ID](https://support.google.com/googleplay/android-developer/answer/6048248?hl=en)** and the iOS Identifier for Advertisers. 
+In the rest of the post we call these random pseudonyms as **deviceID** since we use them as an identifier of a smart device (mobile, pad, etc...).
+
+In particular, it is expected that:
+- behind a deviceID there should be a real physical device used by a human person, 
+- when a physical device uses different apps, those should give the same deviceID (unless it is resetted by the device's user which definetively does not occur at a per minute/hour frequency).
+
+The base data for all future investigations can simply be described as a list of visits of deviceID and apps at certain times.
+<p align="center"> <img src="../images/Deviceid-Reliability/table_data_example.png"> </p>
+Quering the dataset via hive or spark we can test if the data looks like what we expect.
+The goal of the study is to identify a set of pattern which lead to select unreliable device identifiers.
```

Comment:
The goal of the study is to identify a set of pattern which allows us to select only reliable device identifiers and filter out those which are not real, either because of technical problems or due to fraudulent behavior.